transcoder.go

```diff
@@ -216,6 +216,14 @@ func (t *Transcoder) registerMethod(handler http.Handler, methodDesc protoreflec
 		methodConf.streamType = connect.StreamTypeUnary
 	}
 
+	// Add the default HTTP POST route for the method.
+	if err := t.addRule(&annotations.HttpRule{
+		Pattern: &annotations.HttpRule_Post{Post: methodPath},
+		Body:    "*",
+	}, methodConf); err != nil {
+		return err
+	}
+	// Add the declared HTTP rules for the method.
```

Comment:
This seems wrong. We always add the default first and we always fail on error. So if the source annotations or separate provided annotations _also_ include the default mapping (but configured explicitly), this will fail. That seems bad.

It also seems bad that a default mapping could otherwise prevent registration of an explicit one. I could see a case (albeit not a good practice) where a user wants to rename/split an RPC in two, and in effect needs to use a route that looks like the default mapping of method A, but have it route to method B. This prevents that.

So I think we'll probably want special knowledge for these rules in the route trie that they are _defaut_ rules. If a conflict is encountered, always discard the default rule and use the explicitly configured one instead. Only report conflicts between two explicitly configured rules, not between an explicitly configured one and a default. Does that make sense?

Reply:
Added an implicit rule which may be overridden, PTAL.